packages/garbo/src/resources/mayam.ts

```diff
@@ -0,0 +1,262 @@
+function isResonanceAvailable(resonance: Resonance): boolean {
+  const usedChoices = yamUses()
+    .split(",")
+    .map((choice) => choice.trim());
+  const resonanceChoices = resonance.rings.split(" ");
+  return resonanceChoices.every((choice) => !usedChoices.includes(choice));
+}
+
+// Filter resonances to only include available ones
+const AVAILABLE_RESONANCES = () => RESONANCE.filter(isResonanceAvailable);
+
+function findBestRingValue():
+  | MayamCalendar.CombinationString
+  | MayamCalendar.Combination {
+  const filteredOptions = AVAILABLE_RING_OPTIONS;
+
+  // Check for the highest value in each ring, filtering by ring
+  const ring1 = maxBy(
+    filteredOptions.filter((option) => option.ring === 1),
+    (option) => option.value(),
+  );
+  const ring2 = maxBy(
+    filteredOptions.filter((option) => option.ring === 2),
+    (option) => option.value(),
+  );
+  const ring3 = maxBy(
+    filteredOptions.filter((option) => option.ring === 3),
+    (option) => option.value(),
+  );
+  const ring4 = maxBy(
+    filteredOptions.filter((option) => option.ring === 4),
+    (option) => option.value(),
+  );
+
+  // Check for the highest value resonance
+  const maxResonance = maxBy(AVAILABLE_RESONANCES(), (resonance) =>
+    findResonanceValue(resonance.rings),
+  );
+
+  // Sum the total ring value and determine the resonance value
+  const totalRingValue =
+    (ring1?.value() || 0) +
+    (ring2?.value() || 0) +
+    (ring3?.value() || 0) +
+    (ring4?.value() || 0);
+  const maxResonanceValue = maxResonance
+    ? findResonanceValue(maxResonance.rings)
+    : 0;
+
+  // If the resonance is worth more than the best possible rings, return the resonance, otherwise return the four rings
+  if (maxResonance && maxResonanceValue > totalRingValue) {
+    return maxResonance.rings as MayamCalendar.CombinationString;
+  } else {
+    const combination: MayamCalendar.Combination = [
+      ring1.choice as MayamCalendar.Ring<0>,
+      ring2.choice as MayamCalendar.Ring<1>,
+      ring3.choice as MayamCalendar.Ring<2>,
+      ring4.choice as MayamCalendar.Ring<3>,
+    ];
+    return combination;
+  }
+}
+
+function summonTask(): GarboTask {
+  return {
+    name: "Mayam Summons",
+    completed: () =>
+      ["yam4", "explosion", "clock"].every((sym) =>
+        get("_mayamSymbolsUsed").includes(sym),
+      ),
+    do: () => {
+      while (
+        !["yam4", "explosion", "clock"].every((sym) =>
+          get("_mayamSymbolsUsed").includes(sym),
+        )
+      ) {
+        const ringOptions =
+          findBestRingValue() as MayamCalendar.CombinationString;
+        const includesFur = ringOptions
+          .split(" ")
+          .some((option) => option.includes("fur"));
+
+        // If we're going to use Fur, determine the best experience familiar and use it before using any rings
+        if (includesFur) {
+          const bestFamiliar = maxBy(
+            getExperienceFamiliars("free"),
+            (f) => f.expectedValue,
```

Comment:
```suggestion
           "expectedvalue",
```
`maxBy` has an overload for this very use case!